haystack/document_stores/types/filter_policy.py

```diff
@@ -36,10 +40,259 @@ def from_str(filter_policy: str) -> "FilterPolicy":
         return policy
 
 
+def is_legacy(filter_item: Dict[str, Any]) -> bool:
+    """
+    Check if the given filter is a legacy filter.
+
+    :param filter_item: The filter to check.
+    :returns: True if the filter is a legacy filter, False otherwise.
+    """
+    return "operator" not in filter_item
```

Comment:
We don't need this - Legacy filters are officially [deprecated](https://github.com/deepset-ai/haystack/pull/8004/files).